src/algebra/monoid_algebra/basic.lean

```diff
@@ -542,14 +542,32 @@ lemma single_one_comm [comm_semiring k] [mul_one_class G] (r : k) (f : monoid_al
 by { ext, rw [single_one_mul_apply, mul_single_one_apply, mul_comm] }
 
 /-- `finsupp.single 1` as a `ring_hom` -/
-@[simps] def single_one_ring_hom [semiring k] [monoid G] : k →+* monoid_algebra k G :=
+@[simps] def single_one_ring_hom [semiring k] [mul_one_class G] : k →+* monoid_algebra k G :=
 { map_one' := rfl,
   map_mul' := λ x y, by rw [single_add_hom, single_mul_single, one_mul],
   ..finsupp.single_add_hom 1}
 
+/--  A monoid homomorphism `f : G →* H` extends to a monoid homomorphism `G →* monoid_algebra k H`,
+mapping `n : G` to `finsupp.single (f x) 1`. -/
+def single_one_right_hom (k : Type*) {H : Type*} [semiring k] [mul_one_class G] [mul_one_class H]
+  (f : G →* H) : G →* monoid_algebra k H :=
+(of k H).comp f
```

Comment:
I don't think this is really that interesting - can you just inline `(of k H).comp f` where you need it?